pkg/iosrc/uri_windows.go

```diff
@@ -17,7 +25,32 @@ func parseBarePath(path string) (URI, bool, error) {
 	if err != nil {
 		return URI{}, false, err
 	}
-	return URI{Scheme: FileScheme, Path: "/" + filepath.ToSlash(path)}, true, nil
+	// absolute file path for windows will start with a volume so preprepend
+	// slash in front of path.
+	return URI{
+		Scheme: FileScheme,
+		Path:   "/" + filepath.ToSlash(path),
+		Host:   host,
+	}, true, nil
+}
+
+// parseUNCPath parses a microsoft windows UNC path. This not a full
+// implementation.
+// See: https://en.wikipedia.org/wiki/Path_(computing)#POSIX_pathname_definition
+func parseUNCPath(path string) (URI, bool, error) {
+	// Trim first two slashes.
+	path = path[2:]
+	path = filepath.ToSlash(path)
+	z := strings.SplitN(path, "/", 2)
+	// if z is nil then we have just the host name
+	if z == nil {
+		return URI{Scheme: FileScheme, Host: path}, true, nil
+	}
+	u := URI{Scheme: FileScheme, Host: z[0], Path: z[1]}
+	if !strings.HasPrefix(u.Path, "/") {
+		u.Path = "/" + u.Path
+	}
+	return u, true, nil
 }
```

Comment:
This implements [RFC 8089's file URI with Authority mapping for UNC strings](https://tools.ietf.org/html/rfc8089#appendix-E.3.1). If we instead use the [file URI with UNC Path mapping](https://tools.ietf.org/html/rfc8089#appendix-E.3.2), we could simplify this file to
```go
package iosrc

import (
	"path/filepath"
)

func parseBarePath(path string) (URI, bool, error) {
	if filepath.VolumeName(path) == "" {
		if scheme, err := getscheme(path); err != nil || scheme != "" {
			return URI{}, false, err
		}
	}
	path, err := filepath.Abs(path)
	if err != nil {
		return URI{}, false, err
	}
	if len(filepath.VolumeName(path)) == 2 {
		// Add leading '/' to paths beginning with a drive letter.
		path = "/" + path
	}
	return URI{Scheme: FileScheme, Path: filepath.ToSlash(path)}, true, nil
}

func (p URI) Filepath() string {
	path := p.Path
	if path[0] == '/' && len(filepath.VolumeName(path[1:])) == 2 {
		// Strip leading '/' from paths beginning with a drive letter.
		path = path[1:]
	}
	return filepath.FromSlash(path)
}
```

Reply:
If it works that would be great. This is much simpler. Let me put this in and see if it passes tests.

Reply:
Works! Let's do this. Nice tip!